src/Workspaces/CSharp/Portable/Classification/ClassificationHelpers.cs

```diff
@@ -162,12 +191,61 @@ private static string GetClassificationForIdentifier(SyntaxToken token)
             {
                 return ClassificationTypeNames.Keyword;
             }
+            else if (token.Parent is ExternAliasDirectiveSyntax externAliasDirectiveSyntax && externAliasDirectiveSyntax.Identifier == token)
+            {
+                return ClassificationTypeNames.NamespaceName;
+            }
+            else if (token.Parent is LabeledStatementSyntax labledStatementSyntax && labledStatementSyntax.Identifier == token)
+            {
+                return ClassificationTypeNames.LabelName;
+            }
             else
             {
                 return ClassificationTypeNames.Identifier;
             }
         }
 
+        public static bool IsStaticallyDeclared(SyntaxToken token)
+        {
+            if (token.Parent is BaseTypeDeclarationSyntax typeDeclaration && typeDeclaration.Identifier == token)
+            {
+                return typeDeclaration.GetModifiers().Any(SyntaxKind.StaticKeyword);
+            }
+            else if (token.Parent is MethodDeclarationSyntax methodDeclaration && methodDeclaration.Identifier == token)
+            {
+                return methodDeclaration.GetModifiers().Any(SyntaxKind.StaticKeyword);
+            }
+            else if (token.Parent is PropertyDeclarationSyntax propertyDeclaration && propertyDeclaration.Identifier == token)
+            {
+                return propertyDeclaration.GetModifiers().Any(SyntaxKind.StaticKeyword);
+            }
+            else if (token.Parent is EnumMemberDeclarationSyntax enumMemberDeclaration && enumMemberDeclaration.Identifier == token)
+            {
+                return false;  // TODO: Since Enum members are always static is it useful to classify them as static?
+            }
+            else if (token.Parent is VariableDeclaratorSyntax variableDeclarator && variableDeclarator.Identifier == token)
+            {
+                var varDecl = variableDeclarator.Parent as VariableDeclarationSyntax;
+                switch (varDecl.Parent)
+                {
+                    case FieldDeclarationSyntax fieldDeclaration:
+                        return fieldDeclaration.GetModifiers().Any(
+                            modifier => modifier.Kind() == SyntaxKind.StaticKeyword || modifier.Kind() == SyntaxKind.ConstKeyword);
+                    case EventFieldDeclarationSyntax aventFieldDeclarationSyntax:
+                        return aventFieldDeclarationSyntax.GetModifiers().Any(SyntaxKind.StaticKeyword);
+                }
+                return false;
+            }
+            else if (token.Parent is EventDeclarationSyntax eventDeclarationSyntax && eventDeclarationSyntax.Identifier == token)
+            {
+                return eventDeclarationSyntax.GetModifiers().Any(SyntaxKind.StaticKeyword);
```

Comment:
so much repetition of the checks.  Why not just use any of our helpers that give you the modifiers, then just check if the modifers contains static?